File: libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/PreKeyResponseItem.java
/**
 * Copyright (C) 2014-2016 Open Whisper Systems
 * <p>
 * Licensed according to the LICENSE file in this repository.
 */

package org.whispersystems.signalservice.internal.push;

import com.fasterxml.jackson.annotation.JsonProperty;

import org.whispersystems.signalservice.api.push.SignedPreKeyEntity;

public class PreKeyResponseItem {

  @JsonProperty
  public int deviceId;

  @JsonProperty
  public int registrationId;

  @JsonProperty
  public SignedPreKeyEntity signedPreKey;

  @JsonProperty
  public PreKeyEntity preKey;

  @JsonProperty("pqPreKey")
  public KyberPreKeyEntity kyberPreKey;

  public int getDeviceId() {
    return deviceId;
  }

  public int getRegistrationId() {
    return registrationId;
  }

  public SignedPreKeyEntity getSignedPreKey() {
    return signedPreKey;
  }

  public PreKeyEntity getPreKey() {
    return preKey;
  }

  public KyberPreKeyEntity getKyberPreKey() {
    return kyberPreKey;
  }

}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/PreKeyResponseItem.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.internal.push

import com.fasterxml.jackson.annotation.JsonProperty
import org.whispersystems.signalservice.api.push.SignedPreKeyEntity

class PreKeyResponseItem(
  val deviceId: Int = 0,
  val registrationId: Int = 0,
  val signedPreKey: SignedPreKeyEntity? = null,
  val preKey: PreKeyEntity? = null,
  @JsonProperty("pqPreKey") val kyberPreKey: KyberPreKeyEntity? = null
)
